File: leetcode/code/520.detect_capital.go
package main

import (
    "regexp"
)

var validString = regexp.MustCompile(`\b([a-z]+|[A-Z]+|[A-Z][a-z]+)\b`)

func detectCapitalUse(word string) bool {
    return validString.MatchString(word)
}

func main() {
    if !(detectCapitalUse("USA") && detectCapitalUse("Title") && detectCapitalUse("hello")) {
        println("buggy")
    }

    if detectCapitalUse("wOrld") {
        println("buggy")
    }
}

File: leetcode/code/520.detect_capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        result = self.sameCase(word[1:])

        if len(word) > 1 and word[0].islower() and word[1].isupper() and result:
            return False

        return result

    def sameCase(self, chars):
        return (
            len(chars) == 0 or
            len(list(filter(lambda x: x.islower(), chars))) == 0 or
            len(list(filter(lambda x: x.isupper(), chars))) == 0
        )


if __name__ == "__main__":
    s = Solution()
    assert s.detectCapitalUse("USA") is True
    assert s.detectCapitalUse("Little") is True
    assert s.detectCapitalUse("world") is True
    assert s.detectCapitalUse("hLLo") is False
